### DISCUSSION TITLE
What license is this code base provided under?

### DISCUSSION THREAD

**Discussion by Claudenw** [Pi4J/pi4j#310] [General]
The LICENSE file in the root directory indicates Apache 2 licensing, but some of the source files indicate LGPL 3.  If there are different licenses for different parts that I would expect a LICENSE file in the root of things like plugins.

Can anyone help out here?

**Comment by FDelporte** [comment]
Good remark, but you're the first one to notice ;-)

If I remember correctly, there was a tool - at least in V1 - to automatically update all source files with the correct JavaDoc intro stating license and other information. I think this is also in V2 as they all have this start and end:

```
/*-
 * #%L
...
 * #L%
 */
```

@eitch @savageautomate do you remember why we have both apache and lgpl in the sources? In my opinion it should be apache everywhere, correct?

**Comment by Claudenw** [comment]
There is a tool from Apache called RAT (I am working on it) that will find
the source files with differing licenses.  Then you can decide what to do,
but at least you will know what the source says it is.

On Mon, Nov 20, 2023 at 4:35 PM Frank Delporte ***@***.***>
wrote:

> Good remark, but you're the first one to notice ;-)
>
> If I remember correctly, there was a tool - at least in V1 - to
> automatically update all source files with the correct JavaDoc intro
> stating license and other information. I think this is also in V2 as they
> all have this start and end:
>
> /*-
>  * #%L
> ...
>  * #L%
>  */
>
> @eitch <https://github.com/eitch> @savageautomate
> <https://github.com/savageautomate> do you remember why we have both
> apache and lgpl in the sources? In my opinion it should be apache
> everywhere, correct?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Pi4J/pi4j-v2/discussions/310#discussioncomment-7620913>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AASTVHWRT6HRMMBOUYVIZWLYFN2EPAVCNFSM6AAAAAA7TAUWMGVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TMMRQHEYTG>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
LinkedIn: http://www.linkedin.com/in/claudewarren

**Comment by eitch** [comment]
I am pretty sure that this is an oversight and that Apache V2  is the correct license.
